### PR TITLE
Feat/update profile interests page

### PR DESCRIPTION
**Pull request by themonster2015**
## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/design-system-core/src/components/Pill/index.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const Pill: React.FC<PillProps> = ({
       hover={false}
       label={label}
       size={size}
-      icon={icon}
+      {...(active && { icon: icon })}
```
Comment (didd) [inline ui/design-system-core/src/components/Pill/index.tsx]:
```suggestion
      icon={ active ? icon : null }
```

Comment (themonster2015) [reply ui/design-system-core/src/components/Pill/index.tsx]: Thanks Didd. I was using '' and typescript complained but using null works perfectly 👍

File: ui/apps/profile/src/components/pages/profile-engagement/interests-page.tsx [hunk 2]
```diff
@@ -55,7 +108,22 @@ const InterestsPage: React.FC<RootComponentProps> = props => {
     profileDataReq.data,
   );
 
-  const handleTopicClick = (topic: string) => {
+  const successText = `${t('Successfully subscribed to interest')}!`;
+
+  const handleTopicClick = topic => {
+    if (!ownTopics.find(interest => interest.value === topic.value)) {
+      createInterest
+        .mutateAsync({
```
Comment (didd) [inline ui/apps/profile/src/components/pages/profile-engagement/interests-page.tsx]: Unless you need concurrent mutations  or you absolutely need to use Promise, I don't see the reason for using mutateAsync. You will need to manually catch for errors but with mutate this isn't necessary.  So why not add the following inside the `onSuccess` method of the create interest hook ? :)
```
 setFeedbackMessage(successText);
 setShowFeedback(true);
```

File: ui/apps/profile/src/components/pages/profile-engagement/interests-page.tsx [hunk 3]
```diff
@@ -75,21 +146,78 @@ const InterestsPage: React.FC<RootComponentProps> = props => {
...
+              handleClick: () => {
+                //do something
+              },
+            }}
+            saveButton={{
+              label: 'Save',
+              loading: isProcessing,
+              handleClick: interests =>
+                createInterest
+                  .mutateAsync({
```
Comment (didd) [inline ui/apps/profile/src/components/pages/profile-engagement/interests-page.tsx]: Same as the comment above for `mutateAsync`

File: ui/apps/profile/src/components/pages/profile-engagement/interests-page.tsx (hunk 2 above)
Comment (themonster2015) [reply ui/apps/profile/src/components/pages/profile-engagement/interests-page.tsx]: You are right. It is much cleaner this way :). I was complicating things thinking I need mutateAsync to display different feedback messages to the user.

**Comment by didd** [comment]
The PR looks good to me, but I will delay approving it till the pending discussion you have with Marius is resolved.

**Review by kenshyx — APPROVED** [review]

**Pull request merged**